explainer.md

```diff
@@ -518,6 +518,46 @@ File API.
 JavaScript Date objects will be automatically converted to timestamps thanks to
 [implicit conversions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/valueOf).
 
+## Design Decisions
+
+### Default Paths
+This API defaults cookie paths to `/` for cookie write operations, including deletion/expiration. The implicit relative path-scoping of cookies to `.` has caused a lot of additional complexity for relatively little gain given their security equivalence under the same-origin policy and the difficulties arising from multiple same-named cookies at overlapping paths on the same domain. Cookie paths without a trailing `/` are treated as if they had a trailing `/` appended for cookie write operations. Cookie paths must start with `/` for write operations, and must not contain any `..` path segments. Query parameters and URL fragments are not allowed in paths for cookie write operations.
+
+### URL Behavior
+URLs without a trailing `/` are treated as if the final path segment had been removed for cookie read operations, including change monitoring. Paths for cookie read operations are resolved relative to the default read cookie path.
+
+### Secure Cookies
+This API only allows writing "Secure" cookies. This is intended to prevent unintentional leakage to unsecured connections on the same domain. Furthermore it disallows (to the extent permitted by the browser implementation) creation or modification of `Secure-`flagged cookies from unsecured web origins and enforces special rules for the [__Host-](https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis#section-4.1.3.2) and [__Secure-](https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis#section-4.1.3.1) cookie name prefixes.
+
+### Domain Defaults
+This API defaults cookies to "Domain"-less, which in conjunction with "Secure" provides origin-scoped cookie
+behavior in most modern browsers. When practical the `__Host-` cookie name prefix should be used with these cookies so that cooperating browsers origin-scope them.
+
+### Serializing Expiration Dates
+Serialization of expiration times for non-session cookies in a special cookie-specific format has proven cumbersome,
+so this API allows JavaScript Date objects and numeric timestamps (milliseconds since the beginning of the Unix epoch) to be used instead. The inconsistently-implemented Max-Age parameter is not exposed, although similar functionality is available for the specific case of expiring a cookie.
+
+### HTTP Cookie Header Serialization
+Cookies without U+003D (=) code points in their HTTP Cookie header serialization are treated as having an empty name, consistent with the majority of current browsers. Cookies with an empty name cannot be set using values containing U+003D (=) code points as this would result in ambiguous serializations in the majority of current browsers.
```

Comment:
Regarding "Cookies with an empty name cannot be set..." -- I vaguely remember that we changed Chrome's implementation of this spec to completely disallow writing cookies with empty names.

I think we should lift this behavior into the spec and outright disallow setting cookies with empty names, if we haven't already done so by virtute of WPT.

Reply:
I think currently setting empty names is still allowed ([example](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/web_tests/external/wpt/cookie-store/cookieStore_delete_arguments.tentative.https.any.js;l=163?q=empty%20name&ss=chromium%2Fchromium%2Fsrc:third_party%2Fblink%2Fweb_tests%2Fexternal%2Fwpt%2Fcookie-store%2F&start=1)), and only disallowed if the value has an `=` as it mentions above ([example](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/web_tests/external/wpt/cookie-store/cookieStore_set_arguments.tentative.https.any.js;l=34)). Do you think this should be returning a `TypeError` in the case of an empty name?

Reply:
Thank you for explaining! I defer to the cookie experts who reviewed our implementation.